diff --git a/cmake/FindEclipseClp.cmake b/cmake/FindEclipseClp.cmake
index de3ba99..d5a3cf9 100644
--- a/cmake/FindEclipseClp.cmake
+++ b/cmake/FindEclipseClp.cmake
@@ -6,7 +6,10 @@ function(get_eclipse_flag ECLIPSE_BIN FLAG RESULT)
 	set(${RESULT} ${CMD_OUTPUT} PARENT_SCOPE)
 endfunction()
 
-execute_process(COMMAND /usr/bin/which eclipse OUTPUT_VARIABLE ECLIPSE_BIN RESULT_VARIABLE WHICH_RESULT)
+execute_process(COMMAND /usr/bin/which eclipse-clp OUTPUT_VARIABLE ECLIPSE_BIN RESULT_VARIABLE WHICH_RESULT)
+if (NOT ECLIPSE_BIN)
+  execute_process(COMMAND /usr/bin/which eclipse OUTPUT_VARIABLE ECLIPSE_BIN RESULT_VARIABLE WHICH_RESULT)
+endif()
 
 if (ECLIPSE_BIN)
 	string(STRIP ${ECLIPSE_BIN} ECLIPSE_BIN)
@@ -25,7 +28,7 @@ if (ECLIPSE_BIN)
 			set(ECLIPSE_CFLAGS -I${ECLIPSE_INCDIR})
 			set(ECLIPSE_LDFLAGS "-L${ECLIPSE_LIBDIR} -Wl,-R${ECLIPSE_LIBDIR} -leclipse")
 			set(ECLIPSE_LIBRARY_DIRS ${ECLIPSE_LIBDIR})
-			set(ECLIPSE_INCLUDE_DIRS ${ECLIPSE_INCDIR})
+			set(ECLIPSE_INCLUDE_DIRS /usr/include/eclipse-clp ${ECLIPSE_INCDIR})
 		endif()
 	endif()
 endif()
